test_scripts/Resumption/InteriorVehicleData/commonResumptionsInteriorVD.lua

```diff
@@ -127,4 +138,99 @@ function m.preconditions(isPreloadedUpdate, pCountOfRCApps)
   end
 end
 
+function m.reRegisterApp(pAppId, pCheckResumptionData, pCheckResumptionHMILevel, pResultCode)
+  if not pAppId then pAppId = 1 end
+  if not pResultCode then pResultCode = "SUCCESS" end
+  local mobSession = actions.mobile.createSession(pAppId)
+  mobSession:StartService(7)
+  :Do(function()
+      local params = m.cloneTable(m.getConfigAppParams(pAppId))
+      params.hashID = m.hashId[pAppId]
+      local corId = mobSession:SendRPC("RegisterAppInterface", params)
+      m.getHMIConnection():ExpectNotification("BasicCommunication.OnAppRegistered", {
+          application = { appName = m.getConfigAppParams(pAppId).appName }
+        })
+      :Do(function(_, data)
+          m.setHMIAppId(data.params.application.appID, pAppId)
+        end)
+      mobSession:ExpectResponse(corId, { success = true, resultCode = pResultCode })
+    end)
+  pCheckResumptionData()
+  pCheckResumptionHMILevel(pAppId)
+end
+
+function m.resumptionFullHMILevel(pAppId)
+  m.getHMIConnection():ExpectRequest("BasicCommunication.ActivateApp", { appID = m.getHMIAppId(pAppId) })
+  :Do(function(_, data)
+      m.getHMIConnection():SendResponse(data.id, "BasicCommunication.ActivateApp", "SUCCESS", {})
+    end)
+  m.getMobileSession(pAppId):ExpectNotification("OnHMIStatus",
+    { hmiLevel = "NONE", audioStreamingState = "NOT_AUDIBLE", systemContext = "MAIN" },
+    { hmiLevel = "FULL", systemContext = "MAIN", audioStreamingState = "AUDIBLE" })
+  :Times(2)
+end
+
+function m.checkModuleResumptionData(pModuleType, pModuleId)
+  local requestParams = {
+    moduleType = pModuleType,
+    subscribe = true,
+    moduleId = pModuleId
+  }
+  EXPECT_HMICALL("RC.GetInteriorVehicleData", requestParams)
+  :Do(function(_, data)
+      m.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS",
+        { moduleData = m.state.getActualModuleIVData(pModuleType, pModuleId)})
+    end)
+end
+
+function m.getModuleIdNumber(pModuleType, pModuleId)
+  local moduleNumbers = 2
+  for i = 1, moduleNumbers do
+    local out = m.getModuleControlData(pModuleType, i)
+    if out.moduleId == pModuleId then
+      return i
+    end
+  end
+  m.fail(pModuleId .. " is not found for " .. pModuleType .. " in predefinedInteriorVehicleData")
+end
+
+function m.getModuleControlData(pModuleType, pModuleId)
+  local out = rc.predefined.getModuleControlData(pModuleType, 1)
```

Comment:
I would propose to use rc.state.getActualModuleStateOnHMI instead of rc.predefined..